### ISSUE TITLE
Exception

### ISSUE THREAD

**Issue by SemElVik** [habanoz/tezos-reward-distributor#5]
tz@tz:~/tezos-reward-distributor$ python3 src/main.py -C 34 -M 3 tz1 -D -V
2018-11-19 04:04:24,318 - main - INFO - MainThread - Tezos Reward Distributor is Starting
2018-11-19 04:04:24,318 - main - INFO - MainThread - Current network is MAINNET
2018-11-19 04:04:24,318 - main - INFO - MainThread - Baker address is tz1enYtbOh6w91Q665UfMfxK4YT…..
2018-11-19 04:04:24,318 - main - INFO - MainThread - Key name tz1
2018-11-19 04:04:24,319 - main - INFO - MainThread - --------------------------------------------
2018-11-19 04:04:24,319 - main - INFO - MainThread - Copyright Hüseyin ABANOZ 2018
2018-11-19 04:04:24,319 - main - INFO - MainThread - huseyinabanox@gmail.com
2018-11-19 04:04:24,319 - main - INFO - MainThread - Please leave copyright information
2018-11-19 04:04:24,319 - main - INFO - MainThread - --------------------------------------------
2018-11-19 04:04:24,319 - main - INFO - MainThread - DRY RUN MODE
2018-11-19 04:04:24,319 - main - INFO - MainThread - --------------------------------------------
Not found /home/tz/tezos-client
2018-11-19 04:04:24,319 - main - DEBUG - MainThread - Client command is /home/tz/tezos/tezos-client
2018-11-19 04:04:24,319 - main - DEBUG - MainThread - Producer started
2018-11-19 04:04:24,320 - main - INFO - MainThread - If you want to send emails, populate email.ini file under current working directory.
2018-11-19 04:04:24,320 - main - DEBUG - MainThread - Consumer "consumer0" created
2018-11-19 04:04:30,338 - main - INFO - producer  - Payment cycle is 34
2018-11-19 04:04:30,985 - main - INFO - producer  - Total rewards=6.0
2018-11-19 04:04:30,985 - main - INFO - consumer0 - Starting payments for cycle 34
2018-11-19 04:04:30,987 - main - INFO - producer  - Reward created for cycle 34 address KT1NZ3LtKRCwUsTCJQccMsY5a3o amount 1.785000 fee 0.135000 tz type D
2018-11-19 04:04:30,988 - main - INFO - producer  - Reward created for cycle 34 address KT1B99nz7P38MTx783N7RxkBiTv amount 0.969000 fee 0.073980 tz type D
2018-11-19 04:04:30,989 - main - INFO - producer  - Reward created for cycle 34 address KT19ZPyxt9qmLEo5i8NBFVJebi4e amount 0.200000 fee 0.015340 tz type D
2018-11-19 04:04:30,990 - main - INFO - producer  - Reward created for cycle 34 address KT1JJ7Wdwk1b4ovoJ3JBAVQchEhR amount 0.175000 fee 0.000000 tz type O
2018-11-19 04:04:30,991 - main - INFO - producer  - Reward created for cycle 34 address KT1FHwygopixxLRaBBngmShfTBww amount 0.186000 fee 0.000000 tz type O
2018-11-19 04:04:30,991 - main - INFO - producer  - Reward created for cycle 34 address KT1JJ7Wdwk1b4ovoJ3JBAVQchEhs amount 0.123767 fee 0.000000 tz type F
2018-11-19 04:04:30,992 - main - INFO - producer  - Reward created for cycle 34 address KT1FHwygopixxLRaBBngmShfTBws amount 0.131616 fee 0.000000 tz type F
2018-11-19 04:04:30,992 - main - INFO - producer  - Reward creation done for cycle 34
2018-11-19 04:04:30,992 - main - INFO - producer  - Run mode ONETIME satisfied. Killing the thread ...
2018-11-19 04:04:30,992 - main - INFO - producer  - Producer returning ...
2018-11-19 04:04:31,602 - main - DEBUG - consumer0 - Payment will be done in 1 batches
2018-11-19 04:04:31,603 - main - DEBUG - consumer0 - Payment of a batch started
Command is |/home/tz/tezos/tezos-client rpc get http://127.0.0.1:8732/chains/main/blocks/head/context/contracts/tz1/counter|
Answer is ||
--- Logging error ---
Traceback (most recent call last):
  File "/home/tz/tezos-reward-distributor/src/pay/payment_consumer.py", line 70, in run
    payment_logs = batch_payer.pay(payment_items, self.verbose, dry_run=self.dry_run)
  File "/home/tz/tezos-reward-distributor/src/pay/batch_payer.py", line 52, in pay
    payments_log = self.pay_single_batch_wrap(payment_items_chunk, verbose=verbose, dry_run=dry_run)
  File "/home/tz/tezos-reward-distributor/src/pay/batch_payer.py", line 67, in pay_single_batch_wrap
    return_code, operation_hash = self.pay_single_batch(payment_items, verbose, dry_run=dry_run)
  File "/home/tz/tezos-reward-distributor/src/pay/batch_payer.py", line 90, in pay_single_batch
    counter = parse_json_response(send_request(self.comm_counter, verbose))
  File "/home/tz/tezos-reward-distributor/src/util/rpc_utils.py", line 16, in parse_json_response
    raise Exception("Unknown client response format")
Exception: Unknown client response format

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/logging/__init__.py", line 993, in emit
    msg = self.format(record)
  File "/usr/lib/python3.6/logging/__init__.py", line 839, in format
    return fmt.format(record)
  File "/usr/lib/python3.6/logging/__init__.py", line 576, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.6/logging/__init__.py", line 338, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3.6/threading.py", line 884, in _bootstrap
    self._bootstrap_inner()
  File "/usr/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/home/tz/tezos-reward-distributor/src/pay/payment_consumer.py", line 87, in run
    logger.error("Error at reward payment", e)
Message: 'Error at reward payment'
Arguments: (Exception('Unknown client response format',),)
--- Logging error ---
Traceback (most recent call last):
  File "/home/tz/tezos-reward-distributor/src/pay/payment_consumer.py", line 70, in run
    payment_logs = batch_payer.pay(payment_items, self.verbose, dry_run=self.dry_run)
  File "/home/tz/tezos-reward-distributor/src/pay/batch_payer.py", line 52, in pay
    payments_log = self.pay_single_batch_wrap(payment_items_chunk, verbose=verbose, dry_run=dry_run)
  File "/home/tz/tezos-reward-distributor/src/pay/batch_payer.py", line 67, in pay_single_batch_wrap
    return_code, operation_hash = self.pay_single_batch(payment_items, verbose, dry_run=dry_run)
  File "/home/tz/tezos-reward-distributor/src/pay/batch_payer.py", line 90, in pay_single_batch
    counter = parse_json_response(send_request(self.comm_counter, verbose))
  File "/home/tz/tezos-reward-distributor/src/util/rpc_utils.py", line 16, in parse_json_response
    raise Exception("Unknown client response format")
Exception: Unknown client response format

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/logging/__init__.py", line 993, in emit
    msg = self.format(record)
  File "/usr/lib/python3.6/logging/__init__.py", line 839, in format
    return fmt.format(record)
  File "/usr/lib/python3.6/logging/__init__.py", line 576, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.6/logging/__init__.py", line 338, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3.6/threading.py", line 884, in _bootstrap
    self._bootstrap_inner()
  File "/usr/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/home/tz/tezos-reward-distributor/src/pay/payment_consumer.py", line 87, in run
    logger.error("Error at reward payment", e)
Message: 'Error at reward payment'
Arguments: (Exception('Unknown client response format',),)
2018-11-19 04:04:32,227 - main - DEBUG - consumer0 - Exit signal received. Killing the thread...
2018-11-19 04:04:32,227 - main - INFO - consumer0 - Consumer returning ...
2018-11-19 04:04:35,326 - main - INFO - MainThread - Interrupted.
2018-11-19 04:04:35,326 - main - INFO - MainThread - --------------------------------------------------------
2018-11-19 04:04:35,326 - main - INFO - MainThread - Sensitive operations are in progress!
2018-11-19 04:04:35,326 - main - INFO - MainThread - Please wait while the application is being shut down!
2018-11-19 04:04:35,326 - main - INFO - MainThread - --------------------------------------------------------
tz@tz:~/tezos-reward-distributor$

**Comment by gaia** [comment]
So we should always use the alias? I started with a tz PKH, and got

```
2018-12-06 21:11:44,064 - main - DEBUG - producer  - Trying payments for cycle 47
2018-12-06 21:11:44,691 - main - DEBUG - producer  - checking payment_cycle <= current_cycle - (self.nw_config['NB_FREEZE_CYCLE'] + 1) - self.release_override
2018-12-06 21:11:44,692 - main - DEBUG - producer  - checking 47 <= 53 - (5 + 1) - 0
2018-12-06 21:11:44,692 - main - INFO - producer  - Payment cycle is 47

--- Logging error ---
Traceback (most recent call last):
  File "src/main.py", line 103, in run
    reward_data = reward_api.get_rewards_for_cycle_map(payment_cycle)
  File "/home/tezos/scripts/tezos-tools/tezos-reward-distributor/src/tzscan/tzscan_reward_api.py", line 43, in get_rewards_for_cycle_map
    nb_delegators = self.get_nb_delegators(cycle)[0]
  File "/home/tezos/scripts/tezos-tools/tezos-reward-distributor/src/tzscan/tzscan_reward_api.py", line 37, in get_nb_delegators
    raise Exception('GET /tasks/ {}'.format(resp.status_code))
Exception: GET /tasks/ 504

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/logging/__init__.py", line 994, in emit
    msg = self.format(record)
  File "/usr/lib/python3.6/logging/__init__.py", line 840, in format
    return fmt.format(record)
  File "/usr/lib/python3.6/logging/__init__.py", line 577, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.6/logging/__init__.py", line 338, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3.6/threading.py", line 884, in _bootstrap
    self._bootstrap_inner()
  File "/usr/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "src/main.py", line 136, in run
    logger.error("Error at reward calculation", e)
Message: 'Error at reward calculation'
Arguments: (Exception('GET /tasks/ 504',),)
--- Logging error ---
Traceback (most recent call last):
  File "src/main.py", line 103, in run
    reward_data = reward_api.get_rewards_for_cycle_map(payment_cycle)
  File "/home/tezos/scripts/tezos-tools/tezos-reward-distributor/src/tzscan/tzscan_reward_api.py", line 43, in get_rewards_for_cycle_map
    nb_delegators = self.get_nb_delegators(cycle)[0]
  File "/home/tezos/scripts/tezos-tools/tezos-reward-distributor/src/tzscan/tzscan_reward_api.py", line 37, in get_nb_delegators
    raise Exception('GET /tasks/ {}'.format(resp.status_code))
Exception: GET /tasks/ 504

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.6/logging/__init__.py", line 994, in emit
    msg = self.format(record)
  File "/usr/lib/python3.6/logging/__init__.py", line 840, in format
    return fmt.format(record)
  File "/usr/lib/python3.6/logging/__init__.py", line 577, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.6/logging/__init__.py", line 338, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "/usr/lib/python3.6/threading.py", line 884, in _bootstrap
    self._bootstrap_inner()
  File "/usr/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "src/main.py", line 136, in run
    logger.error("Error at reward calculation", e)
Message: 'Error at reward calculation'
Arguments: (Exception('GET /tasks/ 504',),)
2018-12-06 21:12:49,999 - main - DEBUG - producer  - Trying payments for cycle 47
2018-12-06 21:12:50,623 - main - DEBUG - producer  - checking payment_cycle <= current_cycle - (self.nw_config['NB_FREEZE_CYCLE'] + 1) - self.release_override
2018-12-06 21:12:50,623 - main - DEBUG - producer  - checking 47 <= 53 - (5 + 1) - 0
2018-12-06 21:12:50,624 - main - INFO - producer  - Payment cycle is 47
```

**Comment by SemElVik** [comment]
I had today on two different nodes the same thing.
everything was decided by restarting distributor!

**Comment by habanoz** [comment]
Did you get 504 two times on different nodes? And does it persist even after you restart ?

**Comment by SemElVik** [comment]
after restart i paid.  after the restart, I paid for 1 cycle (47) and then I launched by default now # python3 src/main.py payNode2.     I worked in this mode, but I got 504.
on another node I was paying with a duplicate program. There was also an error 504. At first there was a wrong billing address, I corrected it, but I received the same error.
then I deleted all the .csv files and made the correct billing address. I got it seems the same error (not sure now exactly)., I closed the application and paid for 1 cycle (47) everything worked out.
Now I’m running everywhere by default # python3 src / main.py - wait for the end of the cycle and see.
